pkg/kubelet/metrics/metrics.go

```diff
@@ -101,6 +101,14 @@ var (
 			Help:      "Interval in microseconds between relisting in PLEG.",
 		},
 	)
+	ProberResults = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: KubeletSubsystem,
+			Name:      "probe_result",
+			Help:      "The result of a liveness or readiness probe for a container.",
+		},
+		[]string{"probe_type", "container_name"},
```

Comment:
Container name is insufficient to identify the container. You at least need to include the pod name & namespace. You might also want to PodUID.

Reply:
Where can I get the namespace?

Reply:
I guess I could use a k8s client to grab the namespace for a pod, not sure if you know of an easier way.

Reply:
The worker has a reference to the pod, which gives you pod.Name, pod.Namespace, and pod.UID

Reply:
Done